src/storage/observer_worker.rs

```diff
@@ -87,6 +78,12 @@ where
                     .try_dump_old_blob_indexes(self.dump_sem.clone())
                     .await;
             }
+            OperationType::TryUpdateActiveBlob => {
+                self.try_update_active_blob().await?;
+                self.inner

```

Comment:
I think, `try_dump_old_blob_indexes` only needs to be executed if the active blob has been updated.

Additional question: what if the active blob will never be updated (new timestamp with separate pearl instance will be created earlier)? Is there any background procedure that dumps old blob indices?

Reply:
Changed dump to work only in case of active blob checks.
Regarding blob never updating - no, there are no background processes for dumping indexes.